engine/access/rest/rest_api_server.go

```diff
@@ -0,0 +1,139 @@
+		var handler http.Handler
+		handler = route.HandlerFunc
+		handler = newHandler(handler, route.Name, logger)
+		router.
+			Methods(route.Method).
+			Path(route.Pattern).
+			Name(route.Name).
+			Handler(handler)
+	}
+
+	return &http.Server{
+		Addr:    listenAddress,
+		Handler: router,
+	}
+}
+
+// apiRoutes returns the Gorilla Mux routes for each of the API defined in the rest definition
+// currently, it only supports BlocksIdGet
+func apiRoutes(api *RestAPIHandler) generated.Routes {
+	return generated.Routes{
+		generated.Route{
+			Name:        "AccountsAddressGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/accounts/{address}",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "BlocksGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/blocks",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "BlocksIdGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/blocks/{id}",
+			HandlerFunc: api.BlocksIdGet,
+		},
+
+		generated.Route{
+			Name:        "CollectionsIdGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/collections/{id}",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "ExecutionResultsGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/execution_results",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "ExecutionResultsIdGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/execution_results/{id}",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "ScriptsPost",
+			Method:      strings.ToUpper("Post"),
+			Pattern:     "/v1/scripts",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "TransactionResultsTransactionIdGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/transaction_results/{transaction_id}",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "TransactionsIdGet",
+			Method:      strings.ToUpper("Get"),
+			Pattern:     "/v1/transactions/{id}",
+			HandlerFunc: api.NotImplemented,
+		},
+
+		generated.Route{
+			Name:        "TransactionsPost",
+			Method:      strings.ToUpper("Post"),
+			Pattern:     "/v1/transactions",
+			HandlerFunc: api.NotImplemented,
+		},
+	}
+}
+
+// newHandler creates a request handler which adds a logger interceptor to each request to log the request method, uri,
+// duration and response code
+func newHandler(inner http.Handler, name string, logger zerolog.Logger) http.Handler {
+	apiLogger := logger.With().Str("route_name", name).Logger()
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		respWrite := newResponseWriter(w)
+		inner.ServeHTTP(respWrite, r)
+		apiLogger.Info().Str("method", r.Method).
+			Str("uri", r.RequestURI).
+			Dur("duration", time.Since(start)).
```

Comment:
maybe also include the client id, and response size if possible

Reply:
added client IP and user-agent, but response size seems a little tricky since I the actual writing is done by some underlying writer.